lib/matplotlib/tests/test_legend.py

```diff
@@ -671,3 +671,66 @@ def test_no_warn_big_data_when_loc_specified():
         ax.plot(np.arange(5000), label=idx)
     legend = ax.legend('best')
     fig.draw_artist(legend)  # Check that no warning is emitted.
+
+
+def test_plot_multiple_input_multiple_label():
+    # test ax.plot() with multidimensional input
+    # and multiple labels
+    x = [1, 2, 3]
+    y = [[1, 2],
+         [2, 5],
+         [4, 9]]
+    label_arrays = [['low', 'high'],
+                    ('low', 'high'),
+                    np.array(['low', 'high'])]
+    for label in label_arrays:
```

Comment:
This should be parametrized using pytest. See `@pytest.mark.parametrize`

Reply:
Done!